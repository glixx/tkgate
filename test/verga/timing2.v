module top;
   reg ck;

  specify
    $width(posedge ck, 30);
    $width(negedge ck, 25);
  endspecify

  initial
    $monitor("%4d: ck=%b",$time,ck);

  initial
    begin
        ck = 1'b1;
        #50;
	ck = 1'b0;
        $display("No violation expected at %d",$time);
        #50;

	ck = 1'b1;
        $display("No violation expected at %d",$time);
        #50;
	ck = 1'b0;
        $display("No violation expected at %d",$time);
        #50;

	ck = 1'b1;
        $display("No violation expected at %d",$time);
        #20;
	ck = 1'b0;
        $display("$width violation expected at %d on posedge",$time);
        #80;


	ck = 1'b1;
        $display("No violation expected at %d",$time);
        #90;
	ck = 1'b0;
        $display("No violation expected at %d",$time);
        #10;

	ck = 1'b1;
        $display("$width violation expected at %d on negedge",$time);
        #50;
	ck = 1'b0;
        $display("No violation expected at %d",$time);
        #50;

    end

endmodule

