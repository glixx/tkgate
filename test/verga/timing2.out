   0: ck=1
No violation expected at 50
  50: ck=0
No violation expected at 100
 100: ck=1
No violation expected at 150
 150: ck=0
No violation expected at 200
 200: ck=1
$width violation expected at 220 on posedge
Runtime Error at 220: Timing violation in top[top] $width(posedge ck, 30).
 220: ck=0
No violation expected at 300
 300: ck=1
No violation expected at 390
 390: ck=0
$width violation expected at 400 on negedge
Runtime Error at 400: Timing violation in top[top] $width(negedge ck, 25).
 400: ck=1
No violation expected at 450
 450: ck=0
